File: modules/wem-api/src/main/java/com/enonic/wem/api/query/aggregation/Range.java
package com.enonic.wem.api.query.aggregation;

public abstract class Range
{
    private final String key;

    protected Range( final Builder builder )
    {
        this.key = builder.key;
    }

    public String getKey()
    {
        return key;
    }

    public static DateRange.Builder newDateRange()
    {
        return new DateRange.Builder();
    }

    public static NumericRange.Builder newNumericRange()
    {
        return new NumericRange.Builder();
    }

    public static class Builder<T extends Builder>
    {
        private String key;

        public T key( final String key )
        {
            this.key = key;
            return (T) this;
        }
    }

}

File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/query/aggregation/Range.ts
module api.query.aggregation {

    export class Range {

        private key: string;

        constructor(key?: string) {
            this.key = key;
        }

        public setKey(key: string) {
            this.key = key;
        }

        public getKey(): string {
            return this.key;
        }

        public toRangeJson(): api.query.aggregation.RangeJson {

            if (this.getKey() != null) {
                return {
                    "key": this.getKey()
                };
            } else {
                return {};
            }


        }

    }

}